src/test/java/TicTacToe/BoardTest.java

```diff
@@ -1,37 +1,91 @@
 package TicTacToe;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import java.util.ArrayList;
+import java.util.List;
 
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(MockitoExtension.class)
 public class BoardTest {
+    @Mock
+    private Figures figures;
+    @Mock
+    private TicTacToe ticTacToe;
+    @Mock
+    private Player player;
+    @Mock
+    private GameLoop gameLoop;
+    @Mock
+    private Player activePlayer;
+    @Mock
+    private WinLogic winLogic;
+    @InjectMocks
+    Board board;
+    private static final char CIRCLE = Figures.CIRCLE.getCharacter();
+
     @Test
-    public void shouldMarkPosition() {
+    public void shouldFindFreePositions() {
         //given
-        Board board = new Board();
-        Position position = new Position(1, 1);
-        char symbol = Figures.CROSS.getCharacter();
+        List<Position> filledPositions = fillFirstColumnOfBoard(board);
+        int totalPositions = 9;
 
-        // when
-        board.markPosition(position, symbol);
+        //when
+        List<Position> freePositions = board.findFreePositions();
 
-        // then
-        char[][] boardCopy = board.getBoardCopy();
-        assertEquals(symbol, boardCopy[position.getRowNumber()][position.getColumnNumber()]);
+        //then
+        assertFalse(freePositions.isEmpty());
+        assertEquals(totalPositions - filledPositions.size(), freePositions.size());
+
+        for (Position position : filledPositions) {
+            assertFalse(freePositions.contains(position));
+        }
+    }
+
+    @Test
+    public void shouldMarkMove() {
+        //given
+        Position move = new Position(0, 0);
+
+        //when
+        board.markMove(move, CIRCLE);
+
+        //then
+        assertFalse(board.isFreeAtPosition(move));
     }
 
     @Test
-    public void shouldThrowException() {
+    public void shouldReturnOriginalBoard() {
         //given
-        Board board = new Board();
-        Position position = new Position(1, 1);
-        char symbol = Figures.CROSS.getCharacter();
+        Position move = new Position(0, 0);
 
-        // when
-        board.markPosition(position, symbol);
+        //when
+        board.markMove(move, CIRCLE);
```

Comment:
Cot stuejsz w tym teście? Dla mnie testujesz czy `board.getBoardCopy()` zwraca to samo, co oryginalny borad więc to `board.getBoardCopy()` jest `when`

Reply:
Wyżej odpowiedziałem